splice_machine/src/main/java/com/splicemachine/derby/iapi/sql/execute/OperationManagerImpl.java

```diff
@@ -36,8 +38,18 @@
     private ConcurrentMap<UUID, RunningOperation> operations = new ConcurrentHashMap();
     private ConcurrentMap<String, RunningOperation> drdaOperations = new ConcurrentHashMap();
 
+    @Override
+    public RunningOperation getRunningOperation(UUID uuid)
+    {
+        return operations.get(uuid);
+    }
+
     public UUID registerOperation(SpliceOperation operation, Thread executingThread, Date submittedTime, DataSetProcessor.Type engine, String rdbIntTkn) {
         Random rnd = ThreadLocalRandom.current();
+        // Question is if we really need two different UUIDs here.
+        // seems like we don't always have a rdbIntTkn, but shouldn't we maybe only then create a custom UUID?
+        // like if(rdbIntTkn == null) rdbIntTkn = new UUID(rnd.nextLong(), rnd.nextLong()).toString(); ?
+        // HOWEVER, it might be that rdbIntTkn is not really unique, but only per connection?
```

Comment:
If I remember correctly you are right, rdbIntTkn is per connection.

Reply:
ok, comment removed